pkg/operator/operator_test.go

```diff
@@ -158,15 +135,10 @@ func TestOperatorSync_NoOp(t *testing.T) {
 			}
 
 			stopCh := make(<-chan struct{})
-			optr := newFakeOperator(nil, []runtime.Object{infra}, stopCh)
-			optr.imagesFile = imagesFilePath
-
-			err = optr.sync("test-key")
-			if !assert.NoError(t, err, "unexpected sync failure") {
-				t.Fatal()
-			}
+			optr := newFakeOperator(nil, []runtime.Object{newFeatureGate(v1.TechPreviewNoUpgrade), infra}, stopCh)
+			go optr.Run(2, stopCh)
```

Comment:
The problem I have with this is the design of `Run()` itself.

That looks like:

```go
// Run runs the machine config operator.
func (optr *Operator) Run(workers int, stopCh <-chan struct{}) {
	defer utilruntime.HandleCrash()
	defer optr.queue.ShutDown()

	glog.Info("Starting Machine API Operator")
	defer glog.Info("Shutting down Machine API Operator")

	if !cache.WaitForCacheSync(stopCh,
		optr.deployListerSynced,
		optr.featureGateCacheSynced) {
		glog.Error("Failed to sync caches")
		return
	}
	glog.Info("Synced up caches")
	for i := 0; i < workers; i++ {
		go wait.Until(optr.worker, time.Second, stopCh)
	}

	<-stopCh
}
```

- The call to `WaitForCacheSync` can fail, yet we ignore the error.
- Callers of `operator.Run()` can silently fail
- Perhaps the cache in some runs is not sync'd and therefore flakes?

Reply:
Agree that optr.Run() should be returning the error ideally. Though we have not seen "Failed to sync caches" error log in any of those flakes.

This PR is for the flake mentioned in the description. Lets create another PR for returning the error.

Reply:
> This PR is for the flake mentioned in the description. Lets create another PR for returning the error.

Please can we hold for the moment.